sysl2/sysl/ints_view.go

```diff
@@ -0,0 +1,587 @@
+		v.drawSystemView(viewParams, params, nameMap)
+	} else {
+		for _, dep := range params.integrations {
+			appA := dep.Self.Name
+			appB := dep.Target.Name
+			apps := &CallsDrawn{
+				Self:   appA,
+				Target: appB,
+			}
+			var direct []string
+			if _, ok := params.drawableApps[appA]; ok {
+				direct = append(direct, appA)
+			}
+			if _, ok := params.drawableApps[appB]; ok {
+				direct = append(direct, appB)
+			}
+			_, ok := callsDrawn[*apps]
+			if appA != appB && !ok {
+				if direct != nil || viewParams.indirectArrowColor != ArrowColorNone {
+					indirect := ""
+					if direct == nil {
+						indirect = " <<indirect>>"
+					}
+					fmt.Fprintf(
+						v.stringBuilder,
+						"%s --> %s%s",
+						v.VarManagerForComponent(appA, nameMap),
+						v.VarManagerForComponent(appB, nameMap),
+						indirect,
+					)
+					fmt.Fprintln(v.stringBuilder)
+					callsDrawn[*apps] = struct{}{}
+				}
+			}
+		}
+		for _, app := range params.apps {
+			for _, mixin := range v.mod.Apps[app].GetMixin2() {
+				mixinName := strings.Join(mixin.Name.Part, " :: ")
+				fmt.Fprintf(
+					v.stringBuilder,
+					"%s <|.. %s",
+					v.VarManagerForComponent(mixinName, nameMap),
+					v.VarManagerForComponent(app, nameMap),
+				)
+				fmt.Fprintln(v.stringBuilder)
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) drawSystemView(viewParams viewParams, params *IntsParam, nameMap map[string]string) {
+	callsDrawn := map[CallsDrawn]struct{}{}
+	for _, dep := range params.integrations {
+		appA := dep.Self.Name
+		appB := dep.Target.Name
+		apps := &CallsDrawn{
+			Self:   appA,
+			Target: appB,
+		}
+		var direct []string
+		if _, ok := params.drawableApps[appA]; ok {
+			direct = append(direct, appA)
+		}
+		if _, ok := params.drawableApps[appB]; ok {
+			direct = append(direct, appB)
+		}
+		appA = strings.Split(appA, " :: ")[0]
+		appB = strings.Split(appB, " :: ")[0]
+		_, ok := callsDrawn[*apps]
+		if appA != appB && !ok {
+			if direct != nil || viewParams.indirectArrowColor != ArrowColorNone {
+				indirect := ""
+				if direct == nil {
+					indirect = " <<indirect>>"
+				}
+				fmt.Fprintf(
+					v.stringBuilder,
+					"%s --> %s%s",
+					v.VarManagerForComponent(appA, nameMap),
+					v.VarManagerForComponent(appB, nameMap),
+					indirect,
+				)
+				fmt.Fprintln(v.stringBuilder)
+				callsDrawn[*apps] = struct{}{}
+			}
+		}
+	}
+}
+
+func (v *IntsDiagramVisitor) generateComponentView(args *Args, viewParams viewParams, params *IntsParam) string {
+
+	v.stringBuilder.WriteString(`@startuml
+`)
+	if viewParams.diagramTitle != "" {
+		fmt.Fprintln(v.stringBuilder, "title "+viewParams.diagramTitle)
+	}
+	v.stringBuilder.WriteString(`hide stereotype
+scale max 16384 height
+skinparam component {
+  BackgroundColor FloralWhite
+  BorderColor Black
+  ArrowColor Crimson
+`)
+	if viewParams.highLightColor != "" {
+		fmt.Fprintln(v.stringBuilder, "  BackgroundColor<<highlight>> "+viewParams.highLightColor)
```

Comment:
The following statement will work like so:
`fmt.Fprintln(v.stringBuilder, "  BackgroundColor<<highlight>> "+viewParams.highLightColor)`

First a new String will be created after adding two strings (Const string + var String)
then passed to string builder that will copy it to its slice

I think this is better:
```go
v.stringBuilder("  BackgroundColor<<highlight>> ")
v.stringBuilder(viewParams.highLightColor)
v.stringBuilder("\n")
```

Reply:
please change all such calls to as shown above.

Reply:
changed